src/naive_bayes/categorical.rs

```diff
@@ -269,4 +269,32 @@ mod tests {
             vec![0., 0., 1., 1., 1., 0., 1., 0., 1., 1., 0., 1., 1., 1.]
         );
     }
+
+    #[test]
+    fn serde() {
+        let x = DenseMatrix::<f32>::from_2d_array(&[
+            &[3., 4., 0., 1.],
+            &[3., 0., 0., 1.],
+            &[4., 4., 1., 2.],
+            &[4., 2., 4., 3.],
+            &[4., 2., 4., 2.],
+            &[4., 1., 1., 0.],
+            &[1., 1., 1., 1.],
+            &[0., 4., 1., 0.],
+            &[0., 3., 2., 1.],
+            &[0., 3., 1., 1.],
+            &[3., 4., 0., 1.],
+            &[3., 4., 2., 4.],
+            &[0., 3., 1., 2.],
+            &[0., 4., 1., 2.],
+        ]);
+
+        let y = vec![0., 0., 1., 1., 1., 0., 1., 0., 1., 1., 1., 1., 1., 0.];
+        let cnb = CategoricalNB::fit(&x, &y, Default::default()).unwrap();
+
+        let deserialized_cnb: CategoricalNB<f32, DenseMatrix<f32>> =
```

Comment:
This works fine, but as long as I change this to f64 it fails due to floating floating precision

Reply:
That's because derived implementation of PartialEq compares floats directly. Instead, you should compare floats up to some precision, e.g. 
```
        let a:f64 = 10.0 + 1e-9;
        let b:f64 = 10.0 + 1e-9;
        let tolerance = std::f64::EPSILON;
        // comparing a and be using == would be wrong, because float is an approximation of a real number
        (a - b).abs() <= tolerance;
```
check out how I [implement PartialEq for RidgeRegression](https://github.com/smartcorelib/smartcore/blob/development/src/linear/ridge_regression.rs#L110)